### ISSUE TITLE
Keeping track of jobs

### ISSUE THREAD

**Issue by miomao656** [yigit/android-priority-jobqueue#245]
Hi,

First thanks for this great library! :)

I have an issue when adding jobs from widgets like recycler view on bind view i need to know if job for that view is already added. This is an issue due to the fact that anytime you scroll down the list it will add job from item again and again and call onCanceled in job. 
Does the library keep a reference to jobs in queue somewhere and how can i implement this if possible? 
This is needed to avoid adding same jobs again and again and restarting or canceling job in progress that way.

**Comment by yigit** [comment]
Creating a job from RV's onBind may not be a great idea (because it will happen a lot) but you can use [`singleInstanceBy`](http://yigit.github.io/android-priority-jobqueue/javadoc/com/birbit/android/jobqueue/Params.html#singleInstanceBy-java.lang.String-) parameter in the Job so that JobManager will keep only 1 instance of the job at a given time. Make sure you use `addJobInBackground` while adding the job.

**Comment by miomao656** [comment]
Yes i am creating and adding jobs just as how you suggested. So there is no way to retrieve list of jobs from the library so that i can compare if they are already added?

**Comment by yigit** [comment]
There is a `getJobStatus` API but you cannot call it on the main thread because it will possibly wait for a lock on queues depending on JobManager's state.
FYI, It cannot distinguish between a Job that has never been added and a job that has finished.

http://yigit.github.io/android-priority-jobqueue/javadoc/com/birbit/android/jobqueue/JobManager.html#getJobStatus-java.lang.String-
